clients/drcachesim/tools/invariant_checker.cpp

```diff
@@ -410,8 +410,8 @@ invariant_checker_t::parallel_shard_memref(void *shard_data, const memref_t &mem
             }
         }
         // Check if we have any invariant violations caused by PC discontinuities.
```

Comment:
Now that I look at the prior code and the `// Invariant: ` comment indicating what we're checking up on line 375: I would expect the whole decoding sequence and identifying the branch target to be in the helper function.  (There will be some conflict with Nahome's decoding changes, but based purely on this diff, all this decoding is used only for this invariant so I would expect it in the function for this invariant).